index.js

```diff
@@ -10,9 +10,317 @@ function ask(questionText) {
+      if (
+        Number(response).toString() !== "NaN" &&
+        Number(response) % 1 === 0 &&
+        response >= min
+      ) {
+        max = Number(response);
+      } else if (
+        Number(response).toString() !== "NaN" &&
+        Number(response) % 1 !== 0
+      ) {
+        j = 1;
+      } else if (response < min) {
+        j = 2;
+      } else {
+        j = 3;
+      }
+      while (j > 0) {
+        if (j === 1) {
+          response = await ask(
+            "Whoops! That's not an integer. No decimals or fractions please! Enter a new maximum.\n"
+          );
+        } else if (j === 2) {
+          response = await ask(
+            "Whoops! The maximum you just chose is less than the minimum you chose. Please enter a new maximum.\n"
+          );
+        } else if (j === 3) {
+          response = await ask(
+            "Whoops! That's not a number that I can recognize. Please enter a number as a maximum!\n"
+          );
+        } else {
+          console.log("Fatal error: unhandled exception :(");
+          process.exit();
+        }
+        if (
+          Number(response).toString() !== "NaN" &&
+          Number(response) % 1 === 0 &&
+          response >= min
+        ) {
+          max = Number(response);
+          j = 0;
+        } else if (
+          Number(response).toString() !== "NaN" &&
+          Number(response) % 1 !== 0
+        ) {
+          j = 1;
+        } else if (response < min) {
+          j = 2;
+        } else {
+          j = 3;
+        }
+      }
+
+      console.log(
+        "Awesome! I will now attempt to guess your number, which is between " + min + " and " + max + "."
+      );
+      let cutMeInHalf = (max - min) / 2; //I wanted a mathematically "pure" decimal value that would perfectly divide itself in half every iteration without rounding.
+      let guess = Math.round(cutMeInHalf) + min; //of course, the guess has to be an integer, hence "round"
+      let addMe = 0; //we'll add a rounded value of cutMeInHalf every iteration to get right in the center of the possible values to guess with optimum efficiency.
+      tries = 1;
+      guessMax = max + 1; //guessMax and guessMin are values that are "out of bounds; the user has already been told that the value is less/greater than these values. 1 is added/subtracted because max and min themselves are still in play
+      guessMin = min - 1;
+      while (true) {
+        response = await ask(
+          "Is your number... " + guess + "? (Y=Yes; L=No, my number is lower; H=No, my number is higher)\n"
+        );
+        if (response.toLowerCase() == "y") { 
+          if (tries === 1) {
+            console.log(
+              "Wow! I guessed your number in just one try. See ya! B^)"
+            );
+          } else {
+            console.log(
+              "Yay! I guessed your number in " + tries + " tries. Bye!"
+            );
+          }
+          process.exit();
+        } else if (response.toLowerCase() == "l") {
```

Comment:
It is generally better to use === to compare values, since === will only return true if the values are exactly the same, whereas == just checks if they're equivalent, so it's less precise.